src/SFTPClient.cpp

```diff
@@ -182,7 +182,7 @@ void SFTPClient::deref(ExceptionSink* xsink) {
 }
 
 int SFTPClient::sftpConnectedUnlocked() {
-   return (sftp_session ? 1: 0);
+   return (sftp_session ? sftpIsAlive(100,0): 0);
```

Comment:
this needs to call `sftpIsAliveUnlocked()` - the current `SFTPClient::sftpIsAlive()` should be renamed to `SFTPClient::sftpIsAliveUnlocked()` and a new ``SFTPClient::sftpIsAlive()` function needs to be added that grabs the lock and calls `sftpIsAliveUnlocked()` in the lock.